test/integration/change-streams/change_stream.test.js

```diff
@@ -187,6 +187,76 @@ describe('Change Streams', function () {
   });
   afterEach(async () => await mock.cleanup());
 
+  context('ChangeStreamCursor options', function () {
+    let client, db, collection;
+
+    beforeEach(async function () {
+      client = this.configuration.newClient();
+      await client.connect();
+      db = client.db('db');
+      collection = db.collection('collection');
+    });
+
+    afterEach(async function () {
+      await client.close();
+      client = undefined;
+      db = undefined;
+      collection = undefined;
+    });
+
+    context('fullDocument', () => {
+      it('sets fullDocument to `undefined` if no value is passed', function () {
+        const changeStream = new ChangeStream(client);
+
+        expect(changeStream.cursor).to.haveOwnProperty('pipeline');
+        const pipelineOptions = changeStream.cursor.pipeline[0].$changeStream;
+        expect(pipelineOptions).to.haveOwnProperty('fullDocument').to.be.undefined;
```

Comment:
Tests look good to me as is, so no need to change, but I've tried to get in the habbit of using the nested feature when splunking deep into an object, at least when there are questions of nullishness

`expect(changeStream).to.have.nested.property('cursor.pipeline[0].$changeStream.fullDocument', undefined);`

If anything in the chain is nullish we crash without a useful assertion printout, YMMV depending, maybe this case is overkill.

Reply:
I went with this approach - the only concern I have is that when checking for `undefined`, any typo in the string will cause a false negative in the test

Reply:
It shouldn't be the case, I tried the following:
```js
expect({a: {}}).to.have.nested.property('a.b', undefined); // throws
expect({a: {b: undefined}}).to.have.nested.property('a.b', undefined); // passes
```

so actually I think two of your tests need to be changed to say "expect to NOT have nested property" distinct from a defined property that is undefined